containers/README.md

```diff
@@ -0,0 +1,69 @@
+# Containers for checks
+
+Wraps podman to run checks in containers.
+
+## How it works
+
+The `run.sh` script creates a container with a `/restraint` mount point
+that targets restraint's source directory in the user's host.
+The positional arguments passed are run as a command inside the
+container. If no arguments are passed, an interactive shell is executed
+to work inside the container.
+
+## Prerequisites
+
+Ensure that you are able to run podman rootless. As a regular user,
+
+```
+$ podman run --rm -ti fedora:latest
+```
+
+## How to use
+
+From restraint source directory,
+
+```
+$ containers/run.sh make clean check valgrind
+```
+
+The arguments, `make clean check valgrind`, are executed as command
+inside the container, where `/restraint` is the working directory.
+
+The container image can be specified using the `-i` option,
+
+```
+$ containers/run.sh -i fedora:rawhide make clean check valgrind`
+```
+
+The default image is `fedora:latest`.
+
+The container is provisioned automatically with the corresponding
+script in `containers/provision/image.sh`, where image is the
+name of the container image.
+
+Alternative provision scripts can be used with the `-p` option,
+
+```
+$ containers/run.sh -i fedora:rawhide -p /restraint/containers/provision/script.sh
+```
+
+The path for the script is specified within the container.
+
+Containers are persisted and reused for each run. Provisioning only
+takes place if the container doesn't exist or if the `-p` option is
+used.
+
+Containers are named as `restraint-checks-repository-tag`, where
+repository and tag are the ones from the image used for the container.
+For a container using `fedora:latest` image,
```

Comment:
Would be nice to say how user can check if an image is already created in the environment